<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Homework 2 &mdash; Genome Sciences 541 2014.04.17 documentation</title>
    
    <link rel="stylesheet" href="../_static/default.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '2014.04.17',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="top" title="Genome Sciences 541 2014.04.17 documentation" href="../index.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li><a href="../index.html">Genome Sciences 541 2014.04.17 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="homework-2">
<h1>Homework 2<a class="headerlink" href="#homework-2" title="Permalink to this headline">¶</a></h1>
<p>Homework 2 for Genome Sciences 541, assigned by <a class="reference external" href="http://research.fhcrc.org/bloom/en.html">Jesse Bloom</a>.</p>
<p>This homework is due on June 9, 2015.</p>
<div class="section" id="what-you-need">
<h2>What you need<a class="headerlink" href="#what-you-need" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>You first need to have completed <a class="reference external" href="https://github.com/jbloom/GenomeSciences541/tree/gh-pages/source/homework_1">homework 1</a>.</li>
<li>You need a program that can be used to compute matrix exponentials. One such program is <tt class="docutils literal"><span class="pre">scipy</span></tt>, see for example <a class="reference external" href="http://docs.scipy.org/doc/scipy-0.13.0/reference/generated/scipy.linalg.expm.html">http://docs.scipy.org/doc/scipy-0.13.0/reference/generated/scipy.linalg.expm.html</a></li>
</ul>
</div>
<div class="section" id="question-1">
<h2>Question 1<a class="headerlink" href="#question-1" title="Permalink to this headline">¶</a></h2>
<p>Compute the maximum likelihood Jukes-Cantor distance from the starting sequence as a function of the number of simulation steps for your simulations in <a class="reference external" href="https://github.com/jbloom/GenomeSciences541/tree/gh-pages/source/homework_1">homework 1</a>.
Do this for the full sequences and the first, second, and third codon positions separately.
Recall that the estimate of the Jukes-Cantor distance is <span class="math">\(t = \frac{3}{4\mu} \ln\left(\frac{3}{4 p - 1}\right)\)</span> where <span class="math">\(p\)</span> is the pairwise sequence identity. We will use units of time such that <span class="math">\(\mu = 1\)</span>.
If the Jukes-Cantor estimate is accurate, then the estimated distance should scale linearly with the number of simulation steps. Does this happen? Why or why not?</p>
</div>
<div class="section" id="question-2-background">
<h2>Question 2 (background)<a class="headerlink" href="#question-2-background" title="Permalink to this headline">¶</a></h2>
<p>The Kimura 2-paramer model adds a single parameter <span class="math">\(\kappa\)</span> to the Jukes Cantor model. The transition matrix <span class="math">\(\mathbf{P}\)</span> for this model is as follows, where nucleotides are indexed in alphabetical order (A, C, G, T) and matrix entry <span class="math">\(P_{ij}\)</span> is the rate going from <span class="math">\(i\)</span> to <span class="math">\(j\)</span>. Note that <span class="math">\(\mu\)</span> has been defined such that most of the matrix elements are one.</p>
<div class="math">
\[\begin{split}\mathbf{P} = \left[\begin{array}{cccc}
             -\left(2 + \kappa\right) &amp; 1 &amp; \kappa &amp; 1 \\
             1 &amp; -\left(2 + \kappa\right) &amp; 1 &amp; \kappa \\
             \kappa &amp; 1 &amp; -\left(2 + \kappa\right) &amp; 1 \\
             1 &amp; \kappa &amp; 1 &amp; -\left(2 + \kappa\right) \\
             \end{array} \right]\end{split}\]</div>
</div>
<div class="section" id="question-2-more-background">
<h2>Question 2 (more background)<a class="headerlink" href="#question-2-more-background" title="Permalink to this headline">¶</a></h2>
<p>With the definition on the previous slide, let <span class="math">\(\mathbf{p_0}\)</span> be the initial probability distribution. The probability distribution after time <span class="math">\(t\)</span> is then <span class="math">\(\mathbf{p}\left(t\right) = \mathbf{p_0} \exp\left(t \mathbf{P}\right)\)</span> where we are using units of time so that <span class="math">\(\mu = 1\)</span>.</p>
<p>So to get the probability that a site changes from A to C in time <span class="math">\(t\)</span>, set <span class="math">\(\mathbf{p_0} = \left(1, 0, 0, 0\right)\)</span> and then compute element 2 of <span class="math">\(\mathbf{p}\left(t\right)\)</span>. The probability the site changes from A to G is likewise element 3 of <span class="math">\(\mathbf{p}\left(t\right)\)</span>, etc.</p>
</div>
<div class="section" id="question-2">
<h2>Question 2<a class="headerlink" href="#question-2" title="Permalink to this headline">¶</a></h2>
<p>Using <span class="math">\(\kappa = 4\)</span> (the value from your simulation), compute the maximum likelihood distance from your initial sequence to the sequence after 100 steps. You no longer have an analytical formula for <span class="math">\(t\)</span> (one is possible but we haven&#8217;t solved for it), so you can just test values of <span class="math">\(t\)</span> taking very small steps to find the maximum value. How does the maximum likelihood with the Kimura 2-parameter model compare to the maximum likelihood for the same pair of sequences using the Jukes-Cantor model? Note that you do <strong>not</strong> need to compare the estimated value of <span class="math">\(t\)</span> &#8211; rather, the comparison is between the maximum likelihood value itself for the two models (i.e. which fits better).</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Homework 2</a><ul>
<li><a class="reference internal" href="#what-you-need">What you need</a></li>
<li><a class="reference internal" href="#question-1">Question 1</a></li>
<li><a class="reference internal" href="#question-2-background">Question 2 (background)</a></li>
<li><a class="reference internal" href="#question-2-more-background">Question 2 (more background)</a></li>
<li><a class="reference internal" href="#question-2">Question 2</a></li>
</ul>
</li>
</ul>

  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/homework_2/homework_2.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li><a href="../index.html">Genome Sciences 541 2014.04.17 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &copy; Copyright 2014, Jesse Bloom.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3a0.
    </div>
  </body>
</html>