{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Homework 1 for Jesse Bloom's lectures in Genome Sciences 541\n",
    "This homework is due on Tuesday, May 17. Please try to finish it before class so that we can go over it in class.\n",
    "\n",
    "Please complete this homework by downloading this iPython notebook and filling in the empty cells that follow each question with code that answers the question and plots showing the answers."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Part 1: Simulation of pairs of sequences under two nucleotide models, and estimation of evolutionary distance between them using the Jukes-Cantor model\n",
    "In the [last lecture](jbloom.github.io/GenomeSciences541/lecture_2.html), we discussed two nucleotide model, the simple Jukes-Cantor and the more complex HKY85 model. We are going to simulate sequence divergence under both models, and then estimate the evolutionary distance (branch length) between them using the simpler Jukes-Cantor model.\n",
    "\n",
    "### Jukes Cantor model\n",
    "The first model was the [Jukes-Cantor model](https://en.wikipedia.org/wiki/Models_of_DNA_evolution#JC69_model_.28Jukes_and_Cantor.2C_1969.29.5B1.5D). This is a very simple model that assumes every nucleotide is likely to mutate into every other nucleotide. Recall that the substitution matrix for the Jukes-Cantor model is\n",
    "$$\\mathbf{P} = \\begin{bmatrix} -1 & 1/3 & 1/3 & 1/3 \\\\ 1/3 & -1 & 1/3 & 1/3 \\\\ 1/3 & 1/3 & -1 & 1/3 \\\\ 1/3 & 1/3 & 1/3 & -1 \\end{bmatrix}$$\n",
    "\n",
    "As described in the lecture, it turns out that the elements of $e^{\\mu t \\mathbf{P}}$ are given by\n",
    "$$\\left[e^{\\mu t \\mathbf{P}}\\right]_{ij} = \n",
    "\\begin{cases} \n",
    "\\frac{1}{4} + \\frac{3}{4} e^{-\\mu t} & \\mbox{if } i = j\\\\\n",
    "\\frac{1}{4} - \\frac{1}{4} e^{-\\mu t} &\\mbox{if } i \\ne j\\\\\n",
    "\\end{cases}$$\n",
    "\n",
    "If we start with a given sequence, the expected identity $f$ to that starting sequence after an elapsed time $\\mu t$ is $f = \\frac{1}{4} + \\frac{3}{4} e^{-\\mu t}$. So given the identity $f$ from the starting sequence, you can estimate $\\mu t$ by solving that equation for $\\mu t$.\n",
    "\n",
    "### HKY85 model\n",
    "The second model was the [HKY85 model](https://en.wikipedia.org/wiki/Models_of_DNA_evolution#HKY85_model_.28Hasegawa.2C_Kishino_and_Yano_1985.29.5B4.5D). This model allows the sequences to have unequal nucleotide frequencies, which we will denote $\\phi_A$, $\\phi_C$, $\\phi_G$, and $\\phi_T$ where $1 = \\sum_x \\phi_x$. This model also allows transitions to occur at a different rate than transversions: the elevation of the transition rate over the transversion rate is denoted by $\\kappa$.  \n",
    "\n",
    "So the substitution matrix for the HKY85 model is:\n",
    "$$\n",
    "\\mathbf{P} = \n",
    "\\begin{bmatrix}\n",
    "* & \\phi_C & \\kappa \\phi_G & \\phi_T \\\\\n",
    "\\phi_A & * & \\phi_G & \\kappa \\phi_T \\\\\n",
    "\\kappa \\phi_A & \\phi_C & * & \\phi_T \\\\\n",
    "\\phi_A & \\kappa \\phi_C & \\phi_G & * \\\\\n",
    "\\end{bmatrix}\n",
    "$$\n",
    "where the $*$ indicates that the diagonal elements are chosen such that $0 = \\sum_j P_{ij}$.\n",
    "\n",
    "### Simulations with `pyvolve`\n",
    "We are going to simulate the evolution using the [pyvolve](https://github.com/sjspielman/pyvolve) software package.\n",
    "\n",
    "[pyvolve](https://github.com/sjspielman/pyvolve) is written by [Stephanie Spielman](http://sjspielman.org/) and [Claus Wilke](http://wilkelab.org/) (for anyone interested, they are giving seminars at the Hutch [on Monday](https://www.fredhutch.org/en/labs/phs/projects/herbold-computational-biology-program/seminars-events.html) and [in July](http://www.cidid.org/events/2016/7/21/cidid-seminar-stephanie-spielman), respectively).\n",
    "\n",
    "There is a manual for [pyvolve](https://github.com/sjspielman/pyvolve) that can be [downloaded here](https://github.com/sjspielman/pyvolve/raw/master/user_manual/pyvolve_manual.pdf). \n",
    "\n",
    "The first thing you need to do is install the program. Assuming you have a reasonable implementation of `Python` with `pip` installed, you should be able to do this with `pip install pyvolve`. \n",
    "\n",
    "First you need to set up the nucleotide models. In the example below, I do this for the Jukes-Cantor model:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import pyvolve\n",
    "\n",
    "# following section 6.1 of pyvolve manual\n",
    "freqs = [0.25, 0.25, 0.25, 0.25] # equilibrium nucleotide frequencies all equal for Jukes Cantor\n",
    "custom_mu = {'AC':1, 'AG':1, 'AT':1, 'CG':1, 'CT':1, 'GT':1} # rates all equal, set to one\n",
    "jukescantor_model = pyvolve.Model(\"nucleotide\", {'mu':custom_mu, 'state_freqs':freqs})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we use [pyvolve](https://github.com/sjspielman/pyvolve) to simulate the evolution along a simple two-taxon tree. Note that the tree is specified below in [Newick format](https://en.wikipedia.org/wiki/Newick_format). The sequence is of length 1000 nucleotides, and we simulate for a range of times going from 0 to 1.9. After each simulation, we compute the pairwise divergence between the two sequences on the tree."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import numpy\n",
    "import Bio.SeqIO # import Biopython, you may need to install this\n",
    "\n",
    "seqlength = 10000 # sequences of 1000 nucleotides\n",
    "\n",
    "times = numpy.arange(0, 2, 0.1) # divergence times\n",
    "\n",
    "actual_identities = []\n",
    "for t in times: # loop over different branch lengths\n",
    "    tree = pyvolve.read_tree(tree='(s1:{0},s2:{0});'.format(t / 2.0)) # as in section 5 of pyvolve manual\n",
    "    partition = pyvolve.Partition(models=hky85_model, size=seqlength) # as in section 7 of pyvolve manual\n",
    "    evolver = pyvolve.Evolver(tree=tree, partitions=partition) # as in section 8 of pyvolve manual\n",
    "    seqfile = 'seqs_{0}.fasta'.format(t)\n",
    "    evolver(seqfile=seqfile) # run evolve specifying output sequence file\n",
    "    seqs = list(Bio.SeqIO.parse(seqfile, 'fasta'))\n",
    "    assert len(seqs) == 2\n",
    "    fracident = len([i for i in range(seqlength) if seqs[0].seq[i] == seqs[1].seq[i]]) / float(seqlength)\n",
    "    actual_identities.append(fracident)  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "After these simulations, the variable `actual_identities` has a list of the actual identities after the simulation for each divergence time in `times`. \n",
    "\n",
    "**Question 1: Use the equation that relates $\\mu t$ to the expected identity $f$ for the Jukes-Cantor model to estimate $\\mu t$ for each simulated timepoint. Plot the estimated divergence times under the Jukes-Cantor model versus the actual times used in the simulation. What does this relationship look like, and is it what you expect?**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.text.Text at 0x10dfd2110>"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYgAAAEPCAYAAABY9lNGAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJzt3XeYVPX1x/H3ASH+ooAdDSjRoLGhiEowthULiPWJKGAF\nQbGgxor6Mz9QY+wasAYLFjSaaDQgqNgWFQEFQVDpNpoYpUhAkYXz++M7yLjOLnd3586d8nk9zzzM\nvXP3znGecc9+2/mauyMiIlJZvaQDEBGR/KQEISIiGSlBiIhIRkoQIiKSkRKEiIhkpAQhIiIZxZog\nzOwhM1toZpOruWagmc00s0lm1jrOeEREJLq4WxCDgQ5VvWhmRwK/cfcdgd7A/THHIyIiEcWaINz9\nbWBxNZccBzyWunYc0MTMmsYZk4iIRJP0GEQzYE7a8bzUORERSVjSCUJERPLUBgm//zxg27Tj5qlz\nP2NmKholIlIL7m61+blctCAs9chkKHA6gJm1A5a4+8KqbuTuemTp0a9fv8RjKKaHPk99lvn6qItY\nWxBm9iRQBmxuZl8A/YCGgLv7IHcfYWadzGwWsBzoEWc8IiISXawJwt1PjnBNnzhjEBGR2tEgdYkq\nKytLOoSios8ze/RZ5g+rax9VrpiZF0qsIiL5wszwPB6kFhGRAqQEISIiGSlBiIhIRkoQIiKSkRKE\niIhkpAQhIiIZKUGIiEhGShAiIpKREoSIiGSkBCEikofefBNuuCHZGJQgRETyyIIFcOqpcMop8Nvf\nJhuLEoSISB6oqIABA6BVK2jeHKZOhc6dk40p6R3lRERK3ltvwfnnw1Zbhee77JJ0RIEShIhIQhYu\nhCuugNdfh9tvhxNPBKtV3dV4qItJRCTHKirgrrtg992haVP4+GM46aT8Sg6gFoSISE6NHh26kzbb\nDEaNgl13TTqiqilBiIjkwFdfQd++8MorcNtt0KVL/rUYKlMXk4hIJYsWwdKlkI1NLFevhnvugd12\ng803D7OTunbN/+QAakGISIn75huYMAHGj1/375IlITmsWgVbbx0e22yz7nnl46ZNoWHDn997zJjQ\nndS4MbzxRhhzKCTak1pESsbixT9PBosWQZs2sM8+sPfe4d8ddoB69WD58jDTaMEC+PLLdY/04wUL\nQvdRkyY/TSArV8I778Ctt0K3bsm1GOqyJ7UShIgUpSVL4P331yWC8ePhP/+BvfZalwj22QdatgzJ\noC7WrAktkfTksXx5WA3duHF2/ntqSwlCREqee0gGjzwCL78cflG3br0uEey9N+y0E9Svn3SkuVWX\nBKExCBEpaAsXwpAhITEsXw7du8Nzz4XVyKWWDLJNLQgRKTg//AAjRsDgwWEtwfHHQ48ecOCBde8u\nKjZqQYhISZg8OSSFJ58MlU579Aith0aNko6sOClBiEheW7QoJITBg8Mg8xlnhNXILVsmHVnxUxeT\niOSdigoYOTIkhVdegU6dQmuhfXuNK9SUZjGJSFGYMyesOn78cdh22zDg3LUrbLJJ0pEVrrokCA3n\niEjivvgCzj0X9twzDEC/8gqMHQvnnKPkkCQlCBFJzGefQe/eYfHaJpvA9Olwxx35XeG0lChBiEjO\nffopnHVWWLy2xRYhMdx4I2y5ZdKRSTrNYhIRFiyAq68OA8AdOsBhh8Gmm2b/fWbPhr/8Bf7979Cl\nNHNm2BdB8pNaECIlbM0auP9+2GOPUJ20Vaswc6hFC/j97+Haa8NYwOrVdXufWbPCgPPvfgfNm4fE\ncP31Sg75TrOYRErURx/B2WeHJPHAAz8tRf399/DWW6Gm0csvw/z5oVXRoUN4NGsW7T1mzIA//zms\ner7gArjoIg0651pez2Iys45mNs3MZphZ3wyvb25mL5rZJDObYmbd445JpJR9/z383/9BWVmoNjp6\n9M/3KdhwQzj88LDz2ZQpYQVzx44hWeyxR7j+ssvCbKPvv//5e0ybBqeeCvvvDzvuGLqW+vVTcig0\nsbYgzKweMAM4FJgPvAd0dfdpadf0AzZ096vMbAtgOtDU3Ssq3UstCJE6GjUqtBp23x0GDozeEki3\nenUonb22dTFlSkgEHTqE6qmDBsGrr4bWwgUXJF/uutTl7UI5M2sH9HP3I1PHVwLu7jenXdMbaOXu\nfcxse+Bld98pw72UIERqadEiuOKK8Av9rrtCcbtsWbwYXnst3Pu998Jey336qD5SvsjnYn3NgDlp\nx3OBtpWueQB4zczmAxsDXWKOSaRkuMPTT8PFF0PnzmHcIdt/0W+6abh3587Zva8kLx+muV4FfODu\nh5jZb4BXzGwPd/9v5Qv79+//4/OysjLKyspyFqRIofnsszCVdO7csD9Cu3ZJRyS5UF5eTnl5eVbu\nlYsupv7u3jF1nKmLaQRwg7uPTh2/BvR19/GV7qUuJpEIKipgwICw8OzSS8NgcoMGSUclScnnLqb3\ngJZm1gJYAHQFulW6ZipwGDDazJoCOwGfxByXSFGaMCEMQm+yCYwZE2YQidRWrAnC3VebWR9gJGFK\n7UPuPjU1MO3uPgi4ERhsZh8ABlzh7ovijEuk2HzzTVihPGQI3HILnH46WK3+ZhRZRwvlRArQV1/B\nm2+GaaujRoXxhhNOCMlB9YwkXd5Oc80mJQgpZV9+GRJBeXn4d/78sPbg4IPDo00bjTNIZkoQIkVm\n7tx1rYNRo8JWmwceuC4htG6tndUkGiUIkQL3+efrWgejRsHSpXDQQesSQqtWSghSO0oQIgVqzhy4\n/PKwEvmQQ9YlhF13hXqqtSxZkM/TXEUkg++/D4Xw7rwzlKV4+GH45S+Tjkrkp5QgRHLIPWyWc8kl\nYZvN8eNh++2Tjkoks/UmCDPbH5jk7svN7FSgDTDA3T+PPTqRIjJ1aqhwOm9eqHh62GFJRyRSvSi9\nnPcBK8xsT+BSYDbwWKxRiRSRpUtDi+Ggg+Coo2DSJCUHKQxREkRFanT4OOBud78HUCFfkfVYsyaM\nLey8MyxbFiqpXnSR1itI4YgyBrHMzK4CTgMOTG0CpK+4SDXGjg2b5TRoAMOGwT77JB2RSM1FaUF0\nAVYCZ7r7l0Bz4NZYoxIpUAsWQPfuoezFhRfC228rOUjhWm+CSCWFZ4FfpE59DTwXZ1AiheaHH+DW\nW8OCtqZNw57Mp52mtQxS2KLMYjoLOBvYDPgNYZe4+wn7TIuUvJdeCmMLLVvCO+/ATj/bMFekMEUZ\ngzifsE3oOAB3n2lmW8UalUgB+OqrkBjefRcGDgwzlESKSZQG8Ep3/2HtgZltAKjmhZQsd3j00dCd\ntO22MGWKkoMUpygtiFFmdjXwP2Z2OHAeMCzesETy06efQu/eobrqiBGw995JRyQSnygtiCuB/wBT\ngN7ACHf/31ijEskzFRVwxx2w775hkdu77yo5SPGL0oK4wN0HAA+sPWFmF6XOiRS9Dz6AXr1g4421\nz7OUligtiDMynOue5ThE8s5338HVV4cWwznnwOuvKzlIaamyBWFm3YCTge3NbGjaS42ARXEHJpKk\nUaPgrLNgjz1g8mTYZpukIxLJveq6mN4BFgBbALennV8GTI4zKJGkLFkCffvC8OFw991w/PFJRySS\nnCoTRKqc9+fAfrkLRyQ5zz0XNu855phQWK9Jk6QjEklWlJXU7YC7gF2AhkB9YLm7N445NpGcmD8/\nFNb78EP4+99DWW4RiTZIfTfQDZgJ/A/QC7gnzqBEcmHWLLjlFthzT9hllzBbSclBZJ1IW466+ywz\nq+/uq4HBZjYRuCre0ESya/XqUIZ76NBQgnvxYjj6aHjttTAYLSI/FSVBrDCzhsAkM7uFMHCtGpVS\nEJYtg5EjQ0IYPhx+9Ss49lh45JFQhlvVVkWqZmGzuGouMGsBfEXYJOhioAlwr7vPij+8n8Th64tV\nBGDOnJAQhg4N1VX32y8MPB9zDLRokXR0IrllZri71epnC+WXrhKEVGXNGpgwISSFYcNCgujUKbQU\njjgCGms6hZSwWBOEmR0NXA+0IHRJGeC5nsWkBCGVzZgBt98ekkLjxiEhHHNMaDFsEGl0TaT4xZ0g\nZgF/AKYk+RtaCULSjRwZdmzr0we6dNEmPSJVqUuCiPJ31lzgQ/12lnxx993w5z/DM8/AgQcmHY1I\n8YqSIPoCL5pZObBy7Ul3vyOuoEQyWbUK/vhHKC8Pg8877JB0RCLFLUqCuB74L7AhYSW1SM4tXgwn\nnQQNGoSS2xp4FolflATxK3ffPfZIRKowc2ZY0NapE9x2G9Svn3REIqUhyjKhEWZ2ROyRiGTw+utw\nwAFw6aVw551KDiK5FCVBnAu8ZGbfmdm3ZrbMzL6N+gZm1tHMppnZDDPrW8U1ZWY20cw+NLM3ot5b\nitvf/gbdusFTT8HZZycdjUjpiXWhnJnVA2YAhwLzgfeAru4+Le2aJoS9J45w93lmtoW7f53hXppI\nVSIqKuCyy+Cll8IaB+3iJlJ7sUxzNbOd3X2ambXJ9Lq7vx/h/m2Bmam9JTCzp4DjgGlp15wMPOvu\n81L3/VlykNKxdCl07bqusN4mmyQdkUjpqm6Q+hLgbH66m9xaDrSPcP9mwJy047mEpJFuJ6BBqmtp\nY2Cguz8e4d5SZGbPDiuh27eHv/5Vq6FFklbdjnJre32PdPfv018zsw2zHEMbQsLZCBhjZmNyXQxQ\nkvXmm2Ea65/+BOefn3Q0IgLRprm+Q/gFvr5zmcwDtks7bp46l24u8HUqCX1vZm8CewI/SxD9+/f/\n8XlZWRllZWURQpB89/DDcOWV8MQTcPjhSUcjUtjKy8spLy/Pyr2qHKQ2s60JXURDCOMEawc5GgP3\nu/vO6725WX1gOmGQegHwLtDN3aemXbMzYUvTjsAvgHFAF3f/uNK9NEhdZFavDonh+efDYPTO6/1G\niUhNxVWLqQPQnfBX/+2sSxDLgKuj3NzdV5tZH2AkYUrtQ+4+1cx6h5d9UGog/GVgMrAaGFQ5OUjx\nmTULLr4Yli+HceNgs82SjkhEKotSzfUEd382R/FUF4daEAWuoiK0FO67DyZOhN69oV+/UD5DROIR\ndzXX5mbWmNByeIAw9nClu4+szRtK6Zk3Dx54AB58MOzodu65Ybe3DbM51UFEsi7KSuoz3f1b4Ahg\nc+A04KZYo5KCt2ZN2LPhD3+AVq3gq69gxAgYPRpOPVXJQaQQRGlBrG2adAIec/ePzKxWzRUpft98\nA4MHhzIZv/xlaC08+ig0apR0ZCJSU1ESxAQzGwlsD1xlZo2ANfGGJYXEPZTgvu++MMZw7LHw2GPQ\nrh3oTwmRwhVlkLoe0Br4xN2XmNnmQDN3n5yLANPi0CB1nlm2DIYMgfvvhxUr4JxzoHt32HzzpCMT\nkbXqMkgdZQzCgV2BC1PHGxE2D5IStWpVKL+93Xbw6qthj4bp08M5JQeR4hGli+leQpdSe+A6wmym\nZ4F9Y4xL8tTKldClS0gSH34IzZolHZGIxCVKgvidu7cxs4kA7r7YzLT1aAlasSLMStp4Y/jHP6Ch\nvgUiRS1KF9OqVMkMBzCzLdEgdclZtgyOOgq22CJs4KPkIFL8oiSIgcBzwFZmdgPwNvCXWKOSvLJk\nCXToAC1bhimrKsMtUhoi7SiXKqh3KGFNxGvpxfZyRbOYkvHNN3DEEbD//mGPhnpR/qQQkbxRl1lM\nsW45mk1KELm3cCEcdhh06gQ33aQ1DSKFKO5prlKC5s6Fgw6Czp2VHERKlXqT5Wc+/RQOPTSUybj8\n8qSjEZGkVNuCMLP6qb2ipUTMmAEHHwyXXKLkIFLqqk0Q7r4aWGNmTXIUjyToww/hkEPCHg19+iQd\njYgkLUoX03+BKWb2CrB87Ul3v7DqH5FCM3EiHHkk3HEHnHxy0tGISD6IkiD+lXpIkRo7Fo47Du69\nF044IeloRCRfRF0H0RDYKXU43d1XxRpV5hg0zTUGo0aFmUqPPBJWSotIcYl1y1EzKwMeBT4jLJTb\n1szOcPc3a/OGkj9GjoRTTgmlMw49NOloRCTfRNkPYgJwsrtPTx3vBPzd3ffOQXzpcagFkUXDhkHP\nnvCvf8EBByQdjYjEJe6Fcg3WJgcAd58BNKjNm0l+eOYZ6NULhg9XchCRqkUZpB5vZg8CQ1LHpwDj\n4wtJ4vTcc2EK68svQ+vWSUcjIvksShfTL4DzgbV/a74F3OPuP8QcW+U41MVUR8OGhZbDiy9CmzZJ\nRyMiuRBrsT4zu8jdB6zvXNyUIOrmpZfg9NPhhRegbdukoxGRXIl7DOKMDOe61+bNJBmvvhqSw/PP\nKzmISHRVjkGYWTfgZGB7Mxua9lIjYFHcgUl2lJdDt27w7LPw+98nHY2IFJLqBqnfARYAWwC3p51f\nBkyOMyjJjrffhhNPhKefDqW7RURqQhsGFamxY+HYY2HIkLAjnIiUpljHIMzsD2Y208yWmtm3ZrbM\nzL6tzZtJbowfH5LDI48oOYhI7UWZxTQLOCaJfagrxaEWRASTJkGHDjBoUCjAJyKlLe5ZTAuTTg4S\nzZQpoWT3vfcqOYhI3UVdSf008Dywcu1Jd1cJ8Dzy8cehO+mOO1SyW0SyI0qCaAysANJ7sx3tEZE3\nZsyAww+HW24JU1pFRLJBs5gK3OzZUFYG114LZ56ZdDQikm/i3g9iQ6AnsBuw4drz7q5fRwn77LOw\nj8M11yg5iEj2RRmkfhzYGugAjAKaExbLRWJmHc1smpnNMLO+1Vy3r5mtMrM/RL13KZszB9q3h8su\ng969k45GRIpRlGmuE919LzOb7O57mFkD4C13b7fem5vVA2YAhwLzgfeAru4+LcN1rwDfAQ9nGgBX\nF9M68+aFbqXzzoOLL046GhHJZ3FPc127//QSM9sdaAJsFfH+bYGZ7v55ah/rp4BMEzAvAJ4Bvop4\n35K1cGHoVurVS8lBROIVJUEMMrNNgWuAocDHwC0R798MmJN2PDd17kdm9ivgeHe/j7DntVThu+/C\nCukTT4S+VXbWiYhkx3oHqd39wdTTN4EdYojhr0D6rzsliQzWrIEzzoCWLeG665KORkRKQXXlvi8B\nlrr7Q5XO9wQauftfI9x/HrBd2nHz1Ll0+wBPmZkRKsceaWar3H1opevo37//j8/LysooKyuLEEJx\n6NcP5s6F118HUwoVkSqUl5dTXl6elXtVOUhtZhOAdqmxg/TzDYHx7r7Hem9uVh+YThikXgC8C3Sr\nqnSHmQ0GhmmQ+qeGDIE//QnGjYOtoo7+iIgQ3zqIDSonBwB3/yH11/56uftqM+sDjCSMdzzk7lPN\nrHd42QdV/pGogZeKt9+GSy6BN95QchCR3KquBTEFOMzdF1Y63xR41d1b5SC+9PctuRbEJ5/A/vuH\nst0dOiQdjYgUorimud4KDDezg82sUepRBrwA3FabN5PoliyBo48OXUtKDiKShGoXypnZkcCVwO6E\n7p+PgJvc/cXchPeTWEqmBbFqFRx1FOy8MwwcmHQ0IlLI6tKCULG+POMeVkh//jkMHQobRKm3KyJS\nhViL9UluDRwYBqZHj1ZyEJFk6VdQHhk+HG6+GcaMgcaNk45GREqdEkSemDwZevQI3UotWiQdjYhI\nhFpMZnaRmTW24CEze9/Mjljfz0l0X34JxxwDd90F7dZbI1dEJDeiFOs7092/JWw5uilwGnBTrFGV\nkO++g+OOg549oUuXpKMREVknSoJYO/rdCXjc3T9CBfWyYm0Bvh13DOsdRETySZQxiAlmNhLYHrjK\nzBoBa+INqzT06xc2/3ntNRXgE5H8EyVB9ARaA5+4+woz2xzoEW9YxW/IEHjiCRg7FjbccP3Xi4jk\nWpQuJgd2BS5MHW8E6FdaHawtwPfCCyrAJyL5K0qCuBfYD+iWOl4G3BNbREXuk0/CjnCPPw677pp0\nNCIiVYvSxfQ7d29jZhMB3H1xak8IqaFvv1UBPhEpHFFaEKtSG/84gJltiQapa8wdevWCgw4KtZZE\nRPJdlBbEQOA5YCszuwHoDGhSZg3dfTfMng2PPZZ0JCIi0USq5mpmOxO2DTXgtaq2DI1TIVdzHTcu\nrJQeOxZ22CHpaESklMRazdXMerr7Q8C0tHM3ufuVtXnDUvPNN2GF9KBBSg4iUliidDGdYGbfu/sT\nAGZ2D5rmGsmaNXD66dC5Mxx/fNLRiIjUTKQEAQw1szVAR2CJu/eMN6zicPPNsHQp3Hhj0pGIiNRc\nlWMQZrZZ2mEj4HlgNPB/AO6+KPbofhpPQY1BlJdDt27w3nvQvHnS0YhIqYply1Ez+5QwtdXS/l3L\n3T2nPeqFlCC+/BL23hseeQQOPzzpaESklGlP6jxSURGSwsEHQ//+SUcjIqUu7llMp2c67+6a0Z9B\nv35hL2mV7xaRQhdlkHrftOcbEtZDvA8oQVQyYkRYCDdhAtSvn3Q0IiJ1U+MuJjPbBHjK3TvGE1KV\n75vXXUxffAFt28Izz8ABByQdjYhIUJcupii1mCpbTtg8SFJ++AFOOgkuu0zJQUSKR5QxiGGkCvUB\n9Ql7Q/wjzqAKzeWXQ9OmcOmlSUciIpI9UcYgbmNdgqgAPnf3efGFVFj++U8YNiyMO2jbUBEpJtWt\ng1jGz9c/kDq3EpgN/K+7vxZrhOviybsxiBkzYP/94aWXwroHEZF8E8s0V3dvVM0b1gd2B55I/Vty\nvvsu7Ax33XVKDiJSnOq0UM7Merv737IYT3XvlVctiJ49Q5J44gl1LYlI/op1oVx1cpUc8s0jj8A7\n74Q6S0oOIlKsVGqjhqZMgfbtQzG+3XZLOhoRkerleh1Eyfr227C3wx13KDmISPFTCyIi91C+u3Hj\nsDuciEghyOsWhJl1NLNpZjbDzPpmeP1kM/sg9XjbzFrFHVNtDB4MH30EAwYkHYmISG7E2oIws3rA\nDEKBv/nAe0BXd0/f37odMNXdl5pZR6C/u7fLcK/EWhDTp4cSGhp3EJFCk88tiLbATHf/3N1XAU8B\nx6Vf4O5j3X1p6nAs0CzmmGpk5crQtXTddUoOIlJa4k4QzYA5acdzqT4B9AJejDWiGrrmGth2Wzjn\nnKQjERHJrTqtg8gmMzsE6AFUWQ+1f9oWbWVlZZSVlcUa08iR8Pe/w6RJWu8gIoWhvLyc8vLyrNwr\n7jGIdoQxhY6p4ysJ+1nfXOm6PYBngY7uPruKe+V0DOKrr2CvvcIGQIcemrO3FRHJqnweg3gPaGlm\nLcysIdAVGJp+gZltR0gOp1WVHHLNHc48E047TclBREpXrF1M7r7azPoAIwnJ6CF3n2pmvcPLPgj4\nE7AZcK+ZGbDK3dvGGdf63H13aEFcf32SUYiIJEsL5SqZPDm0GsaMgZYtY387EZFY5XMXU0FZsSJM\nab39diUHERG1INKcdx4sWaIS3iJSPBIr911Mnn8eXnxRU1pFRNZSCwKYNw/atAlJYr/9YnkLEZFE\naAyiDlavDtNZ+/RRchARSVfyCeLWW6GiAq6+OulIRETyS0l3Mb37LhxzTNg6dLvtsnprEZG8oC6m\nWli2DE4+Ge65R8lBRCSTkm1BnHEGNGwIDzyQtVuKiOQdTXOtoSefhHHjYMKEpCMREclfJdeC+PRT\naNs2lPLea68sBCYiksc0BhFRRUUYd7jqKiUHEZH1KakEce210KQJ/PGPSUciIpL/SmYM4s034cEH\nYeJEqFdSaVFEpHZK4lfl8uXQo0eYsbT11klHIyJSGEpikPrii+Hrr+Hxx7MclIhIntM012qMGQNP\nPQVTpiQdiYhIYSnqLqaVK6FnTxgwALbYIuloREQKS1EniOuvh9/+Fk48MelIREQKT9F2MU2aBIMG\nwQcfaAMgEZHaKMoWREUFnHkm3HwzbLNN0tGIiBSmokwQt90GW24J3bsnHYmISOEqummu06fDAQeE\nPR5+/ev44xIRyWeqxZSyZk2YtdSvn5KDiEhdFVWCuOee8O955yUbh4hIMSiaLqbPPoN99oHRo8PU\nVhERURcT7nDWWXDZZUoOIiLZUhQJYvBgWLQoJAgREcmOgu9imj8fWreGV16BPfdMIDARkTxWsl1M\n7mFAundvJQcRkWwr6FIb//wnzJwJTz+ddCQiIsWnYLuYvv4aWrWC556Ddu0SDExEJI/VpYupYBPE\nqaeGchp33plgUCIiea7kNgwaPjxsBDR5ctKRiIgUr4JLEEuXwjnnwKOPwkYbJR2NiEjxin0Wk5l1\nNLNpZjbDzPpWcc1AM5tpZpPMrHV197viCjjySGjfPp54RUQkiDVBmFk94G6gA7Ab0M3Mdq50zZHA\nb9x9R6A3cH9V93vjDRgxAm69NcagS0R5eXnSIRQVfZ7Zo88yf8TdgmgLzHT3z919FfAUcFyla44D\nHgNw93FAEzNrmulmvXrBffdBkyZxhlwa9D9hdunzzB59lvkj7gTRDJiTdjw3da66a+ZluAYI01mP\nPjqr8YmISBUKaiX1gAFJRyAiUjpiXQdhZu2A/u7eMXV8JeDufnPaNfcDb7j706njacDB7r6w0r0K\nY8GGiEieydd1EO8BLc2sBbAA6Ap0q3TNUOB84OlUQllSOTlA7f8DRUSkdmJNEO6+2sz6ACMJ3VkP\nuftUM+sdXvZB7j7CzDqZ2SxgOdAjzphERCSagim1ISIiuZV3g9TZXlhX6tb3eZrZwWa2xMzeTz2u\nSSLOQmBmD5nZQjOrssiLvpvRrO+z1PeyZsysuZm9bmYfmdkUM7uwiutq9v1097x5EBLWLKAF0ACY\nBOxc6ZojgeGp578DxiYdd74+In6eBwNDk461EB7AAUBrYHIVr+u7mb3PUt/Lmn2eWwOtU883BqZn\n43dnvrUgsrqwTiJ9ngCaABCBu78NLK7mEn03I4rwWYK+l5G5+5fuPin1/L/AVH6+nqzG3898SxBZ\nXVgnkT5PgP1STc7hZrZrbkIrSvpuZpe+l7VgZr8mtM7GVXqpxt/PgqvmKlk3AdjO3Vek6mI9D+yU\ncEwi+l7WgpltDDwDXJRqSdRJvrUg5gHbpR03T52rfM2267lGgvV+nu7+X3dfkXr+ItDAzDbLXYhF\nRd/NLNH3subMbANCcnjc3f+d4ZIafz/zLUH8uLDOzBoSFtYNrXTNUOB0+HGldsaFdQJE+DzT+yDN\nrC1h6vPc7Kk9AAAB2ElEQVSi3IZZUIyq+8b13ayZKj9LfS9r5WHgY3evqihRjb+fedXF5FpYl1VR\nPk+gs5mdC6wCvgO6JBdxfjOzJ4EyYHMz+wLoBzRE380aW99nib6XNWJm+wOnAFPMbCLgwNWEGYy1\n/n5qoZyIiGSUb11MIiKSJ5QgREQkIyUIERHJSAlCREQyUoIQEZGMlCBERCQjJQiRKphZk9RcfMxs\nGzP7R9IxieSS1kGIVCFV9GyYu7dKOBSRROTVSmqRPHMjsIOZvU/YV2MXd29lZmcAxwMbAS2BO4Bf\nEFayfg90cvclZrYDcA+wBbACOMvdZyTw3yFSK+piEqnalcBsd28DXE4oX7DWboQk0Ra4Afg2dd1Y\nUvVugEFAH3ffN/Xz9+UqcJFsUAtCpHbeSFUbXWFmi4EXUuenAK3MbCPg98A/zWxtQboGCcQpUmtK\nECK1szLtuacdryH8f1UPWJxqVYgUJHUxiVRtGdAo9bxG21+6+zLgUzPrvPacme2RxdhEYqcEIVKF\n1P4Do81sMnALPx2D+MmlVZw/FeiZ2jbzQ+DYGMIUiY2muYqISEZqQYiISEZKECIikpEShIiIZKQE\nISIiGSlBiIhIRkoQIiKSkRKEiIhkpAQhIiIZ/T9aZ3pQ+Li4LQAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x10df3b6d0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Compute the Jukes-Cantor estimates of mu*t and plot them here\n",
    "%matplotlib inline\n",
    "import matplotlib.pyplot as plt\n",
    "import math\n",
    "\n",
    "def jc_estimate(f):\n",
    "    t = -math.log((4 * f - 1) / 3.)\n",
    "    return t\n",
    "\n",
    "estimates = map(jc_estimate, actual_identities)\n",
    "\n",
    "plt.plot(times, estimates)\n",
    "plt.xlabel('time')\n",
    "plt.ylabel('Jukes Cantor estimates')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Question 2: Now define the HKY85 model for pyvolve simulations similar to how the Jukes-Cantor was defined above. Perform similar simulations under the HKY85 model using $\\kappa = 4$, $\\phi_A = \\phi_T = 0.35$, and $\\phi_C = \\phi_G = 0.15$, and then compute the estimated divergences using the Jukes-Cantor model. Plot the actual versus estimated divergences. How well does the Jukes-Cantor estimate perform when the HKY85 model is the true one?**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Simulations under HKY85 model, and comparison of actual and estimated distances.\n",
    "# following section 6.1 of pyvolve manual\n",
    "freqs = [0.7, 0.1, 0.1, 0.1] # equilibrium nucleotide frequencies all equal for Jukes Cantor\n",
    "custom_mu = {'AC':1, 'AG':4, 'AT':1, 'CG':1, 'CT':4, 'GT':1} # rates all equal, set to one\n",
    "hky85_model = pyvolve.Model(\"nucleotide\", {'mu':custom_mu, 'state_freqs':freqs})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Part 2: Simulating a more complex phylogenetic tree and analysis with `RAxML`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we are going to simulate a more complex tree phylogeny using `pyvolve`. Everything is exactly as in _Part 1_, except now we read the tree from the file [`NP_tree.newick`, which you can download here](https://github.com/jbloom/GenomeSciences541/blob/gh-pages/source/homework_1/NP_tree.newick). We then simply simulate evolution along this sequence using the branch lengths that it gives you.\n",
    "\n",
    "Although it is not required, you might look at this tree using some viewer you download such as [FigTree](http://tree.bio.ed.ac.uk/software/figtree/).\n",
    "\n",
    "Here is the simulation done for the Jukes-Cantor model to create the file `JukesCantor_simulated_NP_alignment.fasta`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10000\n"
     ]
    }
   ],
   "source": [
    "treefile = 'NP_tree.newick'\n",
    "\n",
    "# We use the jukescantor_model defined above\n",
    "tree = pyvolve.read_tree(file=treefile) # as in section 5 of pyvolve manual\n",
    "partition = pyvolve.Partition(models=hky85_model, size=seqlength) # as in section 7 of pyvolve manual\n",
    "evolver = pyvolve.Evolver(tree=tree, partitions=partition) # as in section 8 of pyvolve manual\n",
    "jc_simulated_alignment = 'JukesCantor_simulated_NP_alignment.fasta'\n",
    "evolver(seqfile=jc_simulated_alignment)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we are going to use [RAxML](http://sco.h-its.org/exelixis/web/software/raxml/index.html) to fit a phylogeny to that alignment. There is a decent chance that [RAxML](http://sco.h-its.org/exelixis/web/software/raxml/index.html) is already installed if you are using a scientific computing cluster. If not, install it (for a Mac, it can be installed using `MacPorts`)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "\n",
      "This is the RAxML Master Pthread\n",
      "\n",
      "This is RAxML Worker Pthread Number: 1\n",
      "\n",
      "\n",
      "This is RAxML version 7.7.6 released by Alexandros Stamatakis on August 29 2013.\n",
      "\n",
      "With greatly appreciated code contributions by:\n",
      "Andre Aberer (HITS)\n",
      "Simon Berger (HITS)\n",
      "Nick Pattengale (Sandia)\n",
      "Wayne Pfeiffer (SDSC)\n",
      "Akifumi S. Tanabe (NRIFS)\n",
      "\n",
      "\n",
      "Alignment has 1381 distinct alignment patterns\n",
      "\n",
      "Proportion of gaps and completely undetermined characters in this alignment: 0.00%\n",
      "\n",
      "RAxML rapid hill-climbing mode\n",
      "\n",
      "Using 1 distinct models/data partitions with joint branch length optimization\n",
      "\n",
      "\n",
      "Executing 1 inferences on the original alignment using 1 distinct randomized MP trees\n",
      "\n",
      "All free model parameters will be estimated by RAxML\n",
      "ML estimate of 25 per site rate categories\n",
      "\n",
      "Likelihood of final tree will be evaluated and optimized under GAMMA\n",
      "\n",
      "GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units\n",
      "\n",
      "Partition: 0\n",
      "Alignment Patterns: 1381\n",
      "Name: No Name Provided\n",
      "DataType: DNA\n",
      "Substitution Matrix: GTR\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "RAxML was called as follows:\n",
      "\n",
      "raxml -s JukesCantor_simulated_NP_alignment.fasta -n jukescantor -m GTRCAT -p 1 -T 2 \n",
      "\n",
      "\n",
      "Partition: 0 with name: No Name Provided\n",
      "Base frequencies: 0.704 0.098 0.099 0.099 \n",
      "\n",
      "Inference[0]: Time 0.951473 CAT-based likelihood -36978.909913, best rearrangement setting 5\n",
      "\n",
      "\n",
      "Conducting final model optimizations on all 1 trees under GAMMA-based models ....\n",
      "\n",
      "Inference[0] final GAMMA-based Likelihood: -42110.527169 tree written to file /Users/jbloom/Documents/courses/GenomeSciences541/source/homework_1/RAxML_result.jukescantor\n",
      "\n",
      "\n",
      "Starting final GAMMA-based thorough Optimization on tree 0 likelihood -42110.527169 .... \n",
      "\n",
      "Final GAMMA-based Score of best tree -42110.527169\n",
      "\n",
      "Program execution info written to /Users/jbloom/Documents/courses/GenomeSciences541/source/homework_1/RAxML_info.jukescantor\n",
      "Best-scoring ML tree written to: /Users/jbloom/Documents/courses/GenomeSciences541/source/homework_1/RAxML_bestTree.jukescantor\n",
      "\n",
      "Overall execution time: 3.691439 secs or 0.001025 hours or 0.000043 days\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# The -T option may or may not be needed depending on whether you have a multi-threaded version of raxml\n",
    "!raxml -s {jc_simulated_alignment} -n jukescantor -m GTRCAT -p 1 -T 2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[RAxML](http://sco.h-its.org/exelixis/web/software/raxml/index.html) is **not** using the Jukes-Cantor model, it is using the more general [GTR model](https://en.wikipedia.org/wiki/Models_of_DNA_evolution#GTR:_Generalised_time-reversible_.28Tavar.C3.A9_1986.29.5B9.5D). So let's look and see if it inferred model parameters close to what we used in our simulations. To do this, we look in `RAxML_info.jukescantor` file:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\r\n",
      "This is RAxML version 7.7.6 released by Alexandros Stamatakis on August 29 2013.\r\n",
      "\r\n",
      "With greatly appreciated code contributions by:\r\n",
      "Andre Aberer (HITS)\r\n",
      "Simon Berger (HITS)\r\n",
      "Nick Pattengale (Sandia)\r\n",
      "Wayne Pfeiffer (SDSC)\r\n",
      "Akifumi S. Tanabe (NRIFS)\r\n",
      "\r\n",
      "\r\n",
      "Alignment has 1381 distinct alignment patterns\r\n",
      "\r\n",
      "Proportion of gaps and completely undetermined characters in this alignment: 0.00%\r\n",
      "\r\n",
      "RAxML rapid hill-climbing mode\r\n",
      "\r\n",
      "Using 1 distinct models/data partitions with joint branch length optimization\r\n",
      "\r\n",
      "\r\n",
      "Executing 1 inferences on the original alignment using 1 distinct randomized MP trees\r\n",
      "\r\n",
      "All free model parameters will be estimated by RAxML\r\n",
      "ML estimate of 25 per site rate categories\r\n",
      "\r\n",
      "Likelihood of final tree will be evaluated and optimized under GAMMA\r\n",
      "\r\n",
      "GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units\r\n",
      "\r\n",
      "Partition: 0\r\n",
      "Alignment Patterns: 1381\r\n",
      "Name: No Name Provided\r\n",
      "DataType: DNA\r\n",
      "Substitution Matrix: GTR\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "RAxML was called as follows:\r\n",
      "\r\n",
      "raxml -s JukesCantor_simulated_NP_alignment.fasta -n jukescantor -m GTRCAT -p 1 -T 2 \r\n",
      "\r\n",
      "\r\n",
      "Partition: 0 with name: No Name Provided\r\n",
      "Base frequencies: 0.704 0.098 0.099 0.099 \r\n",
      "\r\n",
      "Inference[0]: Time 0.951473 CAT-based likelihood -36978.909913, best rearrangement setting 5\r\n",
      "alpha[0]: 1.000000 rates[0] ac ag at cg ct gt: 0.744238 3.102597 0.738676 0.491824 2.743035 1.000000 \r\n",
      "\r\n",
      "\r\n",
      "Conducting final model optimizations on all 1 trees under GAMMA-based models ....\r\n",
      "\r\n",
      "Inference[0] final GAMMA-based Likelihood: -42110.527169 tree written to file /Users/jbloom/Documents/courses/GenomeSciences541/source/homework_1/RAxML_result.jukescantor\r\n",
      "\r\n",
      "\r\n",
      "Starting final GAMMA-based thorough Optimization on tree 0 likelihood -42110.527169 .... \r\n",
      "\r\n",
      "Final GAMMA-based Score of best tree -42110.527169\r\n",
      "\r\n",
      "Program execution info written to /Users/jbloom/Documents/courses/GenomeSciences541/source/homework_1/RAxML_info.jukescantor\r\n",
      "Best-scoring ML tree written to: /Users/jbloom/Documents/courses/GenomeSciences541/source/homework_1/RAxML_bestTree.jukescantor\r\n",
      "\r\n",
      "Overall execution time: 3.691439 secs or 0.001025 hours or 0.000043 days\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!cat RAxML_info.jukescantor"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This file gives the base frequencies and rates inferred by [RAxML](http://sco.h-its.org/exelixis/web/software/raxml/index.html). \n",
    "\n",
    "**Question 3: What are the base frequencies and rates inferred by [RAxML](http://sco.h-its.org/exelixis/web/software/raxml/index.html), and how do they compare to what you expect?**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "_Put your answer to question 3 here_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Question 4: Repeat the simulations and inferences using the HKY85 model that you defined in Part 1. Now what are the base frequencies and rates inferred by [RAxML](http://sco.h-its.org/exelixis/web/software/raxml/index.html), and how do they compare to what you expect and what [RAxML](http://sco.h-its.org/exelixis/web/software/raxml/index.html) inferred when you used the Jukes-Cantor model?**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "_Put your answer to question 4 here_"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
