<!DOCTYPE html>


<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Homework 1 &mdash; Genome Sciences 541 2014.04.17 documentation</title>
    
    <link rel="stylesheet" href="../_static/basic.css" type="text/css" />
    <link rel="stylesheet" href="../_static/styles.css" type="text/css" />
    <link rel="stylesheet" href="../_static/slides.css" type="text/css" />
    

    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '2014.04.17',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="../_static/common.js"></script>
    <script type="text/javascript" src="../_static/slides.js"></script>
    <script type="text/javascript" src="../_static/sync.js"></script>
    <script type="text/javascript" src="../_static/controller.js"></script>
    <script type="text/javascript" src="../_static/init.js"></script>
    
    <link rel="top" title="Genome Sciences 541 2014.04.17 documentation" href="../index.html" /> 
  </head>
  <body>

<section
   id="slide_container"
   class='slides layout-regular'>


  <article class="slide level-1" id="homework-1">
<h1>Homework 1</h1>
<p>Homework 1 for Genome Sciences 541, assigned by <a class="reference external" href="http://research.fhcrc.org/bloom/en.html">Jesse Bloom</a>.</p>
<p>This homework is due on May 5, 2014.</p>

</article>
<article class="slide level-2" id="files-and-programs">
<h2>Files and programs</h2>
<p>You will need the following files and programs:</p>
<blockquote>
<div><ul class="simple">
<li>The protein-coding DNA sequence for sperm whale myoglobin at <tt class="docutils literal"><span class="pre">sperm-whale_myoglobin.fasta</span></tt> at <a class="reference external" href="https://github.com/jbloom/GenomeSciences541/tree/gh-pages/source/homework_1">homework 1</a>.</li>
<li>The crystal structure of sperm whale myoglobin at <tt class="docutils literal"><span class="pre">1MBN_renumbered.pdb</span></tt> at <a class="reference external" href="https://github.com/jbloom/GenomeSciences541/tree/gh-pages/source/homework_1">homework 1</a>. This is PDB file <a class="reference external" href="http://www.pdb.org/pdb/explore/explore.do?structureId=1MBN">1MBN</a> that has been renumbered so that it matches the numbering that you would get if you sequentially numbered the protein encoded by <tt class="docutils literal"><span class="pre">sperm-whale_myoglobin.fasta</span></tt> starting with 1 at the N-terminal methionine. Note that the N-terminal methionine is missing from the structure.</li>
<li>You will need to download <a class="reference external" href="http://foldx.crg.es/">FoldX</a> (you have to create an account to do this). Download the <tt class="docutils literal"><span class="pre">30b6</span></tt> version.</li>
</ul>
</div></blockquote>

</article>
<article class="slide level-2" id="assignment">
<h2>Assignment</h2>
<p>You will simulate the evolution of myoglobin, starting from the sperm whale myoglobin sequence.</p>
<p>The evolution occurs with a transition-transversion ratio of <span class="math">\(\kappa = 4.0\)</span>. All synonymous mutations are neutral. Mutations to stop codons or that eliminate the N-terminal methionine are lethal. Other mutations are neutral if the stability of the mutated protein continues to exceed that of the initial starting sperm whale myoglobin, or deleterious otherwise.</p>
<p>The basic process is as follows:</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first">Start by repairing the structure of the myoglobin protein in <tt class="docutils literal"><span class="pre">1MBN_renumbered.pdb</span></tt> with <a class="reference external" href="http://foldx.crg.es/">FoldX</a> using the <em>RepairPDB</em> command. This will create a new repaired PDB file that you will use for all subsequent analyses. This protein structure is then optimize for FoldX. To do this, you can create a runfile with the following contents:</p>
<div class="highlight-python"><div class="highlight"><pre>&lt;TITLE&gt;FOLDX_runscript;
&lt;JOBSTART&gt;#;
&lt;PDBS&gt;1MBN_renumbered.pdb;
&lt;BATCH&gt;#;
&lt;COMMANDS&gt;FOLDX_commandfile;
&lt;RepairPDB&gt;1MBN_repaired;
&lt;END&gt;#;
&lt;OPTIONS&gt;FOLDX_optionfile;
&lt;END&gt;#;
&lt;JOBEND&gt;#;
&lt;ENDFILE&gt;#;
</pre></div>
</div>
<p>Then run <a class="reference external" href="http://foldx.crg.es/">FoldX</a> with a command like:</p>
<div class="highlight-python"><div class="highlight"><pre>./foldx64MacOs -runfile repair_runfile.txt
</pre></div>
</div>
<p>This will create the repaired PDB file <tt class="docutils literal"><span class="pre">1MBN_repaired.pdb</span></tt> for all subsequent steps.</p>
<p>Note that running <a class="reference external" href="http://foldx.crg.es/">FoldX</a> requires that you have the <tt class="docutils literal"><span class="pre">rotabase.txt</span></tt> file (distributed with <a class="reference external" href="http://foldx.crg.es/">FoldX</a>) in the current directory.</p>
<p>Initialize the &quot;extra stability&quot; of this protein as <span class="math">\(\Delta G_f = 0\)</span></p>
</li>
<li><p class="first">Randomly choose a site in the nucleotide sequence for the myoglobin. Mutate this sequence with <span class="math">\(\kappa = 4.0\)</span>. Then do the following:</p>
<ul>
<li><p class="first">If the mutation is synonymous, accept it.</p>
</li>
<li><p class="first">If the mutation is to a stop codon or eliminate the N-terminal methionine, reject it.</p>
</li>
<li><p class="first">Otherwise use <a class="reference external" href="http://foldx.crg.es/">FoldX</a> to compute the change in stability (<span class="math">\(\Delta \Delta G\)</span>) caused by the mutations. More negative free energies of folding imply greater stability, so stabilizing mutations have <span class="math">\(\Delta \Delta G &lt; 0\)</span>. You can compute the stability by creating the following input file:</p>
<div class="highlight-python"><div class="highlight"><pre>&lt;TITLE&gt;FOLDX_runscript;
&lt;JOBSTART&gt;#;
&lt;PDBS&gt;1MBN_repaired.pdb;
&lt;BATCH&gt;#;
&lt;COMMANDS&gt;FOLDX_commandfile;
&lt;BuildModel&gt;mutant1,individual_list.txt;
&lt;END&gt;#;
&lt;OPTIONS&gt;FOLDX_optionfile;
&lt;END&gt;#;
&lt;JOBEND&gt;#;
&lt;ENDFILE&gt;#;
</pre></div>
</div>
<p>In this file, <tt class="docutils literal"><span class="pre">1MBN_repaired.pdb</span></tt> is the sequence of the starting myoglobin. The file <tt class="docutils literal"><span class="pre">individual_list.txt</span></tt> is the file that contains the mutation. For instance, if we are mutating leucine 3 to alanine, then it has contents:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">L3A</span><span class="p">;</span>
</pre></div>
</div>
<p>Run <a class="reference external" href="http://foldx.crg.es/">FoldX</a> using:</p>
<div class="highlight-python"><div class="highlight"><pre>./foldx64MacOs -runfile mutate_runfile.txt
</pre></div>
</div>
<p>An output file called <tt class="docutils literal"><span class="pre">Average_mutant1</span></tt> will be created. The last line of this file contains the <span class="math">\(\Delta \Delta G\)</span> of the mutation as the third entry:</p>
<div class="highlight-python"><div class="highlight"><pre>1MBN_repaired_1 0   1.82741 0.0879676   0.823098    1.73246 0.382185    -1.26706    3.13049 -0.119406   -1.93645    -0.0983107  0   0   0   -0.885727   -0.603938   -0.0218432  0   0   0   0   -4.44089e-16    0
</pre></div>
</div>
<p>So here, <span class="math">\(\Delta \Delta G = 1.82741\)</span>. This is the effect of making the mutation L3A to the protein in <tt class="docutils literal"><span class="pre">1MBN_repaired.pdb</span></tt>. Now compute the stability of the new protein as <span class="math">\(\Delta G_{\rm{new}} = \Delta G_f + \Delta \Delta G\)</span>. If <span class="math">\(\Delta G_{\rm{new}} &gt; 0\)</span>, then the mutant protein is unstable, and so the mutation is eliminated by selection. In that case, retain the unmutated sequence. Otherwise, the mutant protein is viable, and the mutation fixes. In this case, update the stability to the new value of <span class="math">\(\Delta G_{\rm{new}\)</span> and take the new mutant structure in <tt class="docutils literal"><span class="pre">1MBN_repaired_1.pdb</span></tt> to be the new parent protein (this structure should have <em>L3A</em>).</p>
<p>So by this process, stabilizing mutations will always fix, and destabilizing mutations will sometimes fix (if the parent already has enough stability).</p>
<p>Remember, it is <strong>essential</strong> that after each accepted nonsynonymous mutation you need to update the parental protein structure to be the new one created by <a class="reference external" href="http://foldx.crg.es/">FoldX</a>, and also to update the stability to the new one after taking into account the effect of the mutation.</p>
</li>
</ul>
</li>
<li><p class="first">Repeat this process until the protein sequences have diverged to 60% identity from the initial sperm whale myoglobin protein sequence. Keep track of the number of steps, the number of fixed synonymous mutations, the number of fixed nonsynonymous mutations, the number of rejected nonsynonymous mutations, and the number of fixed transitions versus transversions.</p>
<p>Also keep track of the coding nucleotide sequences at each step. This last part is important. You should have a file giving the coding nucleotide sequence at each step. Sometimes this sequence will change (synonymous or accepted nonsynonymous mutation) and sometimes it will be the same for consecutive steps (rejected mutation). In the next homework, we will use this file.</p>
</li>
</ol>
</div></blockquote>

</article>
<article class="slide level-2" id="questions">
<h2>Questions</h2>
<ol class="arabic simple">
<li>Turn in the script that you use to perform the analysis.</li>
<li>How many steps did it take to reach 60% protein sequence identity (40% sequence divergence) from the initial sperm whale myoglobin sequence? Make a plot of protein sequence divergence versus number of steps.</li>
<li>Make a plot of the stability of the protein as a function of the number of steps. This stability should generally be slightly less than zero, and should fluctuate in this range due to stabilizing and destabilizing mutations.</li>
<li>What was the fraction of all attempted nonsynonymous mutations that fixed? This fraction of nonsynonymous mutations is often referred to as the ratio <span class="math">\(\omega\)</span>.</li>
<li>Make a plot of the sequence divergence at the first, second, and third codon positions versus the number of steps. Does the sequence diverge faster at some codon positions than others? Why?</li>
<li>If all mutations fixed, how would you expect the ratio of transitions to compare to that of transversions? Keep in mind that <span class="math">\(\kappa = 4.0\)</span>, but that there are more transversions than transitions. What was the actual ratio of transitions to transversions? Is this higher or lower than the expectation? If it seems substantially different, do you have any ideas why?</li>
</ol>

</article>


</section>

<section id="slide_notes">

</section>

  </body>
</html>