<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Lecture 1 &mdash; Genome Sciences 541 2014.04.17 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '2014.04.17',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="top" title="Genome Sciences 541 2014.04.17 documentation" href="index.html" />
    <link rel="prev" title="Genome Sciences 541" href="index.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="index.html" title="Genome Sciences 541"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">Genome Sciences 541 2014.04.17 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="lecture-1">
<h1>Lecture 1<a class="headerlink" href="#lecture-1" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li>Genome Sciences 541, <a class="reference external" href="http://research.fhcrc.org/bloom/en.html">Jesse Bloom</a></li>
<li>These slides are at <a class="reference external" href="http://jbloom.github.io/GenomeSciences541/lecture_1.html">http://jbloom.github.io/GenomeSciences541/lecture_1.html</a></li>
<li>Reading for this lecture is <a class="reference download internal" href="_downloads/Zuckerkandl_and_Pauling-1965.pdf"><tt class="xref download docutils literal"><span class="pre">Zuckerkandl</span> <span class="pre">and</span> <span class="pre">Pauling</span> <span class="pre">(1965)</span></tt></a>.
Read pages 97-114 and 147-152 carefully,
skim pages 115-146,
feel free to skip pages 153-end.</li>
<li>The homework for this lecture is at <a class="reference external" href="https://github.com/jbloom/GenomeSciences541/tree/gh-pages/source/homework_1">homework 1</a>.</li>
</ul>
<div class="section" id="lecture-1-the-evolution-of-protein-coding-genes">
<h2>Lecture 1: the evolution of protein-coding genes<a class="headerlink" href="#lecture-1-the-evolution-of-protein-coding-genes" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><a class="reference download internal" href="_downloads/Zuckerkandl_and_Pauling-1965.pdf"><tt class="xref download docutils literal"><span class="pre">Zuckerkandl</span> <span class="pre">and</span> <span class="pre">Pauling</span> <span class="pre">(1965)</span></tt></a>: one of the first molecular studies of protein evolution</li>
<li>Structure imposes many constraints</li>
<li>Very hard to quantitatively understand these constraints</li>
<li>Sequence divergence <span class="math">\(\ne\)</span> Functional divergence</li>
<li>A molecular clock?</li>
</ul>
<div class="section" id="genes-evolve-at-the-level-of-dna-or-rna-for-some-viruses">
<h3>Genes evolve at the level of DNA (or RNA for some viruses)<a class="headerlink" href="#genes-evolve-at-the-level-of-dna-or-rna-for-some-viruses" title="Permalink to this headline">¶</a></h3>
<p>We routinely abstract the complex molecule DNA to strings of letters with little loss of relevant information:</p>
<div class="highlight-python"><div class="highlight"><pre>atg gtg ctc agc gag gga gaa tgg cag ttg gtt ctg cac gtc ...
</pre></div>
</div>
<p><em>It has not escaped our notice that the specific pairing we have postulated immediately suggests a possible copying mechanism for the genetic material.</em> &#8211; <a class="reference external" href="http://www.nature.com/nature/dna50/watsoncrick.pdf">Watson and Crick (1953)</a></p>
<p><em>The translation from nucleic acid to protein proceeds in a sequential fashion according to a systematic code with relatively simple rules.</em> &#8211; <a class="reference external" href="http://www.nobelprize.org/nobel_prizes/medicine/laureates/1968/nirenberg-lecture.pdf">Nirenberg (1968)</a></p>
</div>
<div class="section" id="but-most-selection-is-on-the-protein">
<h3>But most selection is on the protein<a class="headerlink" href="#but-most-selection-is-on-the-protein" title="Permalink to this headline">¶</a></h3>
<p>The properties of proteins are <strong>not</strong> easily abstracted.</p>
<p>Proteins are linear polymers encoded by DNA:</p>
<div class="highlight-python"><div class="highlight"><pre>atg gtg ctc agc gag gga gaa tgg cag ttg gtt ctg cac gtc ...
M   V   L   S   E   G   E   W   Q   L   V   L   H   V   ...
</pre></div>
</div>
<p>Proteins derive their relevant properties from their three-dimensional structures.</p>
</div>
<div class="section" id="properties-of-proteins-are-not-easily-abstracted">
<h3>Properties of proteins are not easily abstracted<a class="headerlink" href="#properties-of-proteins-are-not-easily-abstracted" title="Permalink to this headline">¶</a></h3>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/myoglobin_static.png"><img alt="_images/myoglobin_static.png" src="_images/myoglobin_static.png" style="width: 61%;" /></a>
<p class="caption"><a class="reference external" href="https://github.com/jbloom/GenomeSciences541/tree/gh-pages/source/myoglobin_structure">PyMol analysis of myoglobin</a></p>
<div class="legend">
<em>Perhaps the most remarkable features of the molecule are its complexity and its lack of symmetry. The arrangement seems to be almost totally lacking in the kind of regularities which one instinctively anticipates, and it is more complicated than has been predicated by any theory of protein structure.</em> &#8211; <a class="reference external" href="http://www.chem.uwec.edu/Chem452_Media/pages/readings/media/Kendrew_et_al_1958.pdf">Kendrew et al (1958)</a></div>
</div>
</div>
<div class="section" id="context-for-reading-zuckerkandl-and-pauling-1965">
<h3>Context for reading Zuckerkandl and Pauling (1965)<a class="headerlink" href="#context-for-reading-zuckerkandl-and-pauling-1965" title="Permalink to this headline">¶</a></h3>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/Zuckerkandl_and_Pauling-1965.png"><img alt="_images/Zuckerkandl_and_Pauling-1965.png" src="_images/Zuckerkandl_and_Pauling-1965.png" style="width: 90%;" /></a>
<p class="caption"><a class="reference download internal" href="_downloads/Zuckerkandl_and_Pauling-1965.pdf"><tt class="xref download docutils literal"><span class="pre">Zuckerkandl</span> <span class="pre">and</span> <span class="pre">Pauling</span> <span class="pre">(1965)</span></tt></a></p>
</div>
</div>
<div class="section" id="they-knew-the-structure-of-sperm-whale-myoglobin">
<h3>They knew the structure of sperm whale myoglobin<a class="headerlink" href="#they-knew-the-structure-of-sperm-whale-myoglobin" title="Permalink to this headline">¶</a></h3>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/myoglobin_static.png"><img alt="_images/myoglobin_static.png" src="_images/myoglobin_static.png" style="width: 90%;" /></a>
<p class="caption"><a class="reference external" href="https://github.com/jbloom/GenomeSciences541/tree/gh-pages/source/myoglobin_structure">PyMol analysis of myoglobin</a></p>
<div class="legend">
Structure had been determined by <a class="reference external" href="http://www.chem.uwec.edu/Chem452_Media/pages/readings/media/Kendrew_et_al_1958.pdf">Kendrew et al (1958)</a></div>
</div>
</div>
<div class="section" id="they-knew-that-single-mutations-could-potentially-have-large-biological-consequences">
<h3>They knew that single mutations could potentially have large biological consequences<a class="headerlink" href="#they-knew-that-single-mutations-could-potentially-have-large-biological-consequences" title="Permalink to this headline">¶</a></h3>
<div class="highlight-python"><div class="highlight"><pre><span class="n">MVHLTPEEKSAVT</span><span class="o">...</span>
<span class="n">MVHLTPvEKSAVT</span><span class="o">...</span>
</pre></div>
</div>
<p><a class="reference external" href="http://www.sciencemag.org/content/110/2865/543">&#8220;Sickle Cell Anemia, a Molecular Disease&#8221; (Pauling et al, 1949)</a></p>
<p><a class="reference external" href="http://www.nature.com/nature/journal/v178/n4537/abs/178792a0.html">&#8220;A Specific Chemical Difference between Globins of Normal and Sickle-cell Anemia Hemoglobins&#8221; (Ingram, 1956)</a></p>
</div>
<div class="section" id="they-knew-the-sequences-of-a-variety-of-homologous-globins">
<h3>They knew the sequences of a variety of homologous globins<a class="headerlink" href="#they-knew-the-sequences-of-a-variety-of-homologous-globins" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li>sperm whale myoglobin</li>
<li>human myoglobin</li>
<li>human hemoglobin <span class="math">\(\alpha\)</span>, <span class="math">\(\beta\)</span>, <span class="math">\(\gamma\)</span>, and <span class="math">\(\delta\)</span> chains</li>
<li>horse hemoglobin <span class="math">\(\alpha\)</span> and <span class="math">\(\beta\)</span> chains</li>
<li>cattle hemoglobin <span class="math">\(\alpha\)</span> and <span class="math">\(\beta\)</span> chains</li>
<li>cattle fetal hemoglobin chain</li>
<li>pig hemoglobin <span class="math">\(\alpha\)</span> and <span class="math">\(\beta\)</span> chains</li>
<li>gorilla <span class="math">\(\alpha\)</span> and <span class="math">\(\beta\)</span> chains</li>
<li>carp hemoglobin <span class="math">\(\alpha\)</span> chain</li>
<li>lamprey hemoglobin chain</li>
</ul>
</div>
<div class="section" id="analysis-of-myoglobin-homologs">
<h3>Analysis of myoglobin homologs<a class="headerlink" href="#analysis-of-myoglobin-homologs" title="Permalink to this headline">¶</a></h3>
<p>We will perform an analysis similar to that of Zuckerkandl and Pauling, and look at some of the conclusions that they draw.</p>
</div>
<div class="section" id="myoglobin-homologs">
<h3>Myoglobin homologs<a class="headerlink" href="#myoglobin-homologs" title="Permalink to this headline">¶</a></h3>
<div class="highlight-python"><div class="highlight"><pre>&gt;carp
MA----DHELVLKCWGGVEADFEGTGGEVLTRLFKQHPETQKLFPKFVGIA-QSDLAGNAAVKAHGATVLKSWASCLKARGDHAAILKPLATTHANTHKIALNNFRLITEVLVKVMAEKAGLD--AGGQSALRRVMDVVIGDIDTYYKEIGFAG
&gt;chicken
MGLSDQEWQQVLTIWGKVEADIAGHGHEVLMRLFHDHPETLDRFDKFKGLKTPDQMKGSEDLKKHGATVLTQLGKILKQKGNHESELKPLAQTHATKHKIPVKYLEFISEVIIKVIAEKHAADFGADSQAAMKKALELFRNDMASKYKEFGFQG
&gt;horse
MGLSDGEWQQVLNVWGKVEADIAGHGQEVLIRLFTGHPETLEKFDKFKHLKTEAEMKASEDLKKHGTVVLTALGGILKKKGHHEAELKPLAQSHATKHKIPIKYLEFISDAIIHVLHSKHPGDFGADAQGAMTKALELFRNDIAAKYKELGFQG
&gt;human
MGLSDGEWQLVLNVWGKVEADIPGHGQEVLIRLFKGHPETLEKFDKFKHLKSEDEMKASEDLKKHGATVLTALGGILKKKGHHEAEIKPLAQSHATKHKIPVKYLEFISECIIQVLQSKHPGDFGADAQGAMNKALELFRKDMASNYKELGFQG
&gt;mouse
MGLSDGEWQLVLNVWGKVEADLAGHGQEVLIGLFKTHPETLDKFDKFKNLKSEEDMKGSEDLKKHGCTVLTALGTILKKKGQHAAEIQPLAQSHATKHKIPVKYLEFISEIIIEVLKKRHSGDFGADAQGAMSKALELFRNDIAAKYKELGFQG
&gt;sperm-whale
MVLSEGEWQLVLHVWAKVEADVAGHGQDILIRLFKSHPETLEKFDRFKHLKTEAEMKASEDLKKHGVTVLTALGAILKKKGHHEAELKPLAQSHATKHKIPIKYLEFISEAIIHVLHSRHPGDFGADAQGAMNKALELFRKDIAAKYKELGYQG
&gt;tuna
MA----DFDAVLKCWGPVEADYTTMGGLVLTRLFKEHPETQKLFPKFAGIA-QADIAGNAAISAHGATVLKKLGELLKAKGSHAAILKPLANSHATKHKIPINNFKLISEVLVKVMHEKAGLD--AGGQTALRNVMGIIIADLEANYKELGFSG
&gt;turtle
MGLSDDEWNHVLGIWAKVEPDLTAHGQEVIIRLFQLHPETQERFAKFKNLTTIDALKSSEEVKKHGTTVLTALGRILKQKNNHEQELKPLAESHATKHKIPVKYLEFICEIIVKVIAEKHPSDFGADSQAAMKKALELFRNDMASKYKEFGFLG
&gt;zebrafish
MA----DHDLVLKCWGAVEADYAANGGEVLNRLFKEYPDTLKLFPKFSGIS-QGDLAGSPAVAAHGATVLKKLGELLKAKGDHAALLKPLANTHANIHKVALNNFRLITEVLVKVMAEKAGLD--AAGQGALRRVMDAVIGDIDGYYKEIGFAG
</pre></div>
</div>
<p>Sequences at <a class="reference external" href="https://github.com/jbloom/GenomeSciences541/tree/gh-pages/source/myoglobin_homologs">Analysis of myoglobin homologs</a></p>
</div>
<div class="section" id="site-variability-among-myoglobin-homologs">
<h3>Site variability among myoglobin homologs<a class="headerlink" href="#site-variability-among-myoglobin-homologs" title="Permalink to this headline">¶</a></h3>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/myoglobin_site_preferences_logoplot.jpg"><img alt="_images/myoglobin_site_preferences_logoplot.jpg" src="_images/myoglobin_site_preferences_logoplot.jpg" style="width: 91%;" /></a>
<p class="caption">Code /data to produce this figure at <a class="reference external" href="https://github.com/jbloom/GenomeSciences541/tree/gh-pages/source/myoglobin_homologs">Analysis of myoglobin homologs</a></p>
</div>
</div>
<div class="section" id="very-little-sequence-is-strictly-conserved">
<h3>Very little sequence is strictly conserved<a class="headerlink" href="#very-little-sequence-is-strictly-conserved" title="Permalink to this headline">¶</a></h3>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/invariant_sites.png"><img alt="_images/invariant_sites.png" src="_images/invariant_sites.png" style="width: 90%;" /></a>
<p class="caption"><a class="reference download internal" href="_downloads/Zuckerkandl_and_Pauling-1965.pdf"><tt class="xref download docutils literal"><span class="pre">Zuckerkandl</span> <span class="pre">and</span> <span class="pre">Pauling</span> <span class="pre">(1965)</span></tt></a></p>
</div>
</div>
<div class="section" id="relatively-few-insertions-and-deletions">
<h3>Relatively few insertions and deletions<a class="headerlink" href="#relatively-few-insertions-and-deletions" title="Permalink to this headline">¶</a></h3>
<p><em>Deletions or additions of one to several amino acid residues are expected to be eliminated by natural selection in a high proportion of cases. Those that are preserved should be mostly found at either end of a chain, at the end of helices, in short helices, or in nonhelical regions, notably in loops that may be shortened or lengthened without  affecting the steric relationships in the rest of the molecule, A deletion or addition in the middle of a long helix would result in so many simultaneous alterations in side-chain interactions that it is highly unlikely that the tertiary structure and the function of the molecule could survive such an event. The deletions or additions found in hemoglobin and myoglobin chains are compatible with these generalities.</em></p>
<p><a class="reference download internal" href="_downloads/Zuckerkandl_and_Pauling-1965.pdf"><tt class="xref download docutils literal"><span class="pre">Zuckerkandl</span> <span class="pre">and</span> <span class="pre">Pauling</span> <span class="pre">(1965)</span></tt></a></p>
</div>
<div class="section" id="structure-is-highly-conserved">
<h3>Structure is highly conserved<a class="headerlink" href="#structure-is-highly-conserved" title="Permalink to this headline">¶</a></h3>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/structural_conservation.png"><img alt="_images/structural_conservation.png" src="_images/structural_conservation.png" style="width: 90%;" /></a>
<p class="caption"><a class="reference download internal" href="_downloads/Zuckerkandl_and_Pauling-1965.pdf"><tt class="xref download docutils literal"><span class="pre">Zuckerkandl</span> <span class="pre">and</span> <span class="pre">Pauling</span> <span class="pre">(1965)</span></tt></a></p>
</div>
<p>This observation has stood the test of time. In fact, sequences with as little as 30-35% identity generally have
very similar structures. See <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1166865/?page=1">Chothia and Lesk (1984)</a> and <a class="reference external" href="http://cbio.mskcc.org/publications/papers/sander/038.pdf">Sander and Schneider (1991)</a>.</p>
</div>
<div class="section" id="id1">
<h3>Structure is highly conserved<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/sperm_whale_tuna_myoglobin_overlay_static.png"><img alt="_images/sperm_whale_tuna_myoglobin_overlay_static.png" src="_images/sperm_whale_tuna_myoglobin_overlay_static.png" style="width: 80%;" /></a>
<p class="caption"><a class="reference external" href="https://github.com/jbloom/GenomeSciences541/tree/gh-pages/source/myoglobin_structure">PyMol analysis of myoglobin</a></p>
<div class="legend">
Structural alignment of sperm whale (gray) and yellowfin tuna (orange) myoglobins. These proteins are only 57% identical.</div>
</div>
</div>
<div class="section" id="different-proteins-evolve-at-different-rates">
<h3>Different proteins evolve at different rates<a class="headerlink" href="#different-proteins-evolve-at-different-rates" title="Permalink to this headline">¶</a></h3>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/cytochrome_C.png"><img alt="_images/cytochrome_C.png" src="_images/cytochrome_C.png" style="width: 100%;" /></a>
<p class="caption"><a class="reference download internal" href="_downloads/Zuckerkandl_and_Pauling-1965.pdf"><tt class="xref download docutils literal"><span class="pre">Zuckerkandl</span> <span class="pre">and</span> <span class="pre">Pauling</span> <span class="pre">(1965)</span></tt></a></p>
</div>
</div>
<div class="section" id="zuckerkandl-and-pauling-propose-a-molecular-clock">
<h3>Zuckerkandl and Pauling propose a molecular clock<a class="headerlink" href="#zuckerkandl-and-pauling-propose-a-molecular-clock" title="Permalink to this headline">¶</a></h3>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/conclusions_1.png"><img alt="_images/conclusions_1.png" src="_images/conclusions_1.png" style="width: 100%;" /></a>
<p class="caption"><a class="reference download internal" href="_downloads/Zuckerkandl_and_Pauling-1965.pdf"><tt class="xref download docutils literal"><span class="pre">Zuckerkandl</span> <span class="pre">and</span> <span class="pre">Pauling</span> <span class="pre">(1965)</span></tt></a></p>
</div>
</div>
<div class="section" id="id2">
<h3>Zuckerkandl and Pauling propose a molecular clock<a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h3>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/conclusions_2.png"><img alt="_images/conclusions_2.png" src="_images/conclusions_2.png" style="width: 100%;" /></a>
<p class="caption"><a class="reference download internal" href="_downloads/Zuckerkandl_and_Pauling-1965.pdf"><tt class="xref download docutils literal"><span class="pre">Zuckerkandl</span> <span class="pre">and</span> <span class="pre">Pauling</span> <span class="pre">(1965)</span></tt></a></p>
</div>
</div>
<div class="section" id="important-implications">
<h3>Important implications<a class="headerlink" href="#important-implications" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The extent of sequence change can be used to reconstruct evolutionary relationships (<em>molecular phylogenetics</em>)</p>
</li>
<li><p class="first">However, the constant background sequence change can make it difficult to pinpoint biologically important changes:</p>
<blockquote>
<div><ul class="simple">
<li><em>Positive selection</em> : for instance, take a look at <a class="reference external" href="http://nextflu.org/H3N2/">nextflu</a></li>
<li><em>Disease-causing mutations</em></li>
</ul>
</div></blockquote>
</li>
</ul>
</div>
<div class="section" id="quantitative-analyses-of-the-molecular-evolution-of-protein-coding-genes">
<h3>Quantitative analyses of the molecular evolution of protein-coding genes<a class="headerlink" href="#quantitative-analyses-of-the-molecular-evolution-of-protein-coding-genes" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li>Today we will work on forming a conceptual view of the process molecular evolution</li>
<li>In the homework, you will simulate this process</li>
<li>In the next three lectures, we will discuss approaches to quantitatively analyze this type of data from the perspective of molecular phylogenetics</li>
<li>There is a constant tension between <em>biochemical realism</em> and <em>analytical tractability</em>: many of the assumptions are made not because we think they are strictly true, but because we have to make them to do the analysis.</li>
</ul>
</div>
<div class="section" id="the-evolution-studied-in-molecular-phylogenetics-is-due-to-both-mutation-and-selection">
<h3>The evolution studied in molecular phylogenetics is due to both mutation and selection<a class="headerlink" href="#the-evolution-studied-in-molecular-phylogenetics-is-due-to-both-mutation-and-selection" title="Permalink to this headline">¶</a></h3>
<div class="highlight-python"><div class="highlight"><pre>atg gtg ctc agc gag gga gaa tgg cag ttg gtt ctg cTc gtc ...
atg gtg ctc agc gag gga gaa tgA cag ttg gtt ctg cac gtc ...
atg gtg ctc agc Cag gga gaa tgg cag ttg gtt ctg cac gtc ...
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre>atg gtg Atc agc gag gga gaa tgg cag ttg gtt ctg cac gtc ...
atg gtg ctc agc Cag gga gaa tgg cTg ttg gtt ctg cac gtc ...
atg gtg ctc agc Cag gga gaa tgg cag ttg gtA ctg cac gtc ...
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre>atg gtA ctc agc Cag gga gaa tgg cag ttg gtt ctg cac gtc ...
atg gtg ctc agc Cag gga gaa tgg cTg ttg gtt ctg Aac gtc ...
atg gtg ctc agc Cag gga gaa tgg cTg ttg gtA ctg cac gtc ...
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre>atg atg cAc agc Cag gga gaa tgg cTg ttg gtt ctg cac gtc ...
atg gtg ctc agc Cag gga gaa tgg cTg ttg gtt ctg Aac gtc ...
atg gtg ctc agc Cag gga gaa tgg cTg ttg gtA ctg cac gtc ...
</pre></div>
</div>
</div>
<div class="section" id="we-will-ignore-the-intra-population-polymorphism">
<h3>We will ignore the intra-population polymorphism<a class="headerlink" href="#we-will-ignore-the-intra-population-polymorphism" title="Permalink to this headline">¶</a></h3>
<div class="highlight-python"><div class="highlight"><pre>atg gtg ctc agc cag gga gaa tgg cag ttg gtt ctg cac gtc ...
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre>atg gtg ctc agc Cag gga gaa tgg cag ttg gtt ctg cac gtc ...
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre>atg gtg ctc agc Cag gga gaa tgg cTg ttg gtt ctg cac gtc ...
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre>atg gtg ctc agc Cag gga gaa tgg cTg ttg gta ctg cac gtc ...
</pre></div>
</div>
<p>We ignore within-species variation, and treat the evolving populations as a single clonal sequence. This assumption may sometimes be problematic, as the frequency of an allele in a population is not necessarily equal to its probability of ultimately fixing. We are not going to worry about this problem here...</p>
</div>
<div class="section" id="we-assume-that-selection-is-on-the-protein-sequence">
<h3>We assume that selection is on the protein sequence<a class="headerlink" href="#we-assume-that-selection-is-on-the-protein-sequence" title="Permalink to this headline">¶</a></h3>
<p>This leads to the distinction between <em>synonymous</em> mutations:</p>
<div class="highlight-python"><div class="highlight"><pre>atg gtg ctc agc gag gga gaa tgg cag ttg gtt ctg cac gtc ...
M   V   L   S   E   G   E   W   Q   L   V   L   H   V   ...
atg gtg ctc agc gag ggC gaa tgg cag ttg gtt ctg cac gtc ...
M   V   L   S   E   G   E   W   Q   L   V   L   H   V   ...
</pre></div>
</div>
<p>and <em>nonsynonymous</em> mutations:</p>
<div class="highlight-python"><div class="highlight"><pre>atg gtg ctc agc gag gga gaa tgg cag ttg gtt ctg cac gtc ...
M   V   L   S   E   G   E   W   Q   L   V   L   H   V   ...
atg gtg ctc agc gag Cga gaa tgg cag ttg gtt ctg cac gtc ...
M   V   L   S   E   R   E   W   Q   L   V   L   H   V   ...
</pre></div>
</div>
<p>What would you expect about their relative rates of fixation?</p>
</div>
<div class="section" id="we-assume-that-genes-can-be-aligned-and-ignore-gaps">
<h3>We assume that genes can be aligned, and ignore gaps<a class="headerlink" href="#we-assume-that-genes-can-be-aligned-and-ignore-gaps" title="Permalink to this headline">¶</a></h3>
<div class="highlight-python"><div class="highlight"><pre>atg gtg ctc agc cag gga gaa tgg cag ttg gtt ctg cac gtc ...
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre>atg gtg ctc agc Cag gga gaa tgg cag ttg gtt ctg cac gtc ...
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre>atg gtg ctc agc Cag gga gaa tgg cTg ttg --- ctg cac gtc ...
</pre></div>
</div>
<p>The results are therefore dependent on the alignment. We&#8217;d get a different evolutionary scenario if we aligned the last homolog as:</p>
<div class="highlight-python"><div class="highlight"><pre>atg gtg ctc agc Cag gga gaa tgg cTg ttg CtG cAC --- gtc ...
</pre></div>
</div>
</div>
<div class="section" id="we-envision-mutation-as-identical-across-sites">
<h3>We envision mutation as identical across sites<a class="headerlink" href="#we-envision-mutation-as-identical-across-sites" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">Each nucleotide mutation is equally likely (Jukes-Cantor model)</p>
</li>
<li><p class="first">Some nucleotide mutations are likely than others</p>
</li>
<li><p class="first">On clear observation is that <em>transition</em> mutations tend to be more likely than <em>transversion</em> mutations</p>
<blockquote>
<div><ul class="simple">
<li><em>transitions</em>: purine to purine (<span class="math">\(A \leftrightarrow G\)</span>), pyrimidine to pyrimidine (<span class="math">\(C \leftrightarrow T\)</span>)</li>
<li><em>transversions</em>: purine to pyrimidine or pyrimidine to purine (<span class="math">\(A\leftrightarrow C\)</span>, <span class="math">\(A\leftrightarrow T\)</span>, <span class="math">\(G\leftrightarrow T\)</span>, <span class="math">\(G\leftrightarrow C\)</span>)</li>
<li>The symbol <span class="math">\(\kappa\)</span> is often used to denote the relative rate of given <em>transition</em> mutation to a given <em>transversion</em> mutation. A typical value might be <span class="math">\(\kappa \approx 4\)</span> although this varies among organisms.</li>
</ul>
</div></blockquote>
</li>
</ul>
</div>
<div class="section" id="we-know-that-selection-is-not-identical-across-sites">
<h3>We know that selection is <strong>not</strong> identical across sites<a class="headerlink" href="#we-know-that-selection-is-not-identical-across-sites" title="Permalink to this headline">¶</a></h3>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/myoglobin_site_preferences_logoplot.jpg"><img alt="_images/myoglobin_site_preferences_logoplot.jpg" src="_images/myoglobin_site_preferences_logoplot.jpg" style="width: 91%;" /></a>
<p class="caption">Code /data to produce this figure at <a class="reference external" href="https://github.com/jbloom/GenomeSciences541/tree/gh-pages/source/myoglobin_homologs">Analysis of myoglobin homologs</a></p>
</div>
</div>
<div class="section" id="id3">
<h3>We know that selection is <strong>not</strong> identical across sites<a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h3>
<p>However, as we will see in future lectures, we often are forced to assume that it is identical (or nearly so) across sites.</p>
</div>
<div class="section" id="simulating-this-process-your-homework">
<h3>Simulating this process (your homework)<a class="headerlink" href="#simulating-this-process-your-homework" title="Permalink to this headline">¶</a></h3>
<p>In <a class="reference external" href="https://github.com/jbloom/GenomeSciences541/tree/gh-pages/source/homework_1">homework 1</a>, you will simulate the evolution of myoglobin.</p>
<p>You will start with the coding sequence for sperm whale myoglobin, and introduce random nucleotide mutations. Mutations that alter the protein sequence will be subjected to purifying selection as modeled by the <a class="reference external" href="http://foldx.crg.es/manual1.jsp">FoldX energy function</a>.</p>
<p>The output of these simulations will be set of simulated evolved sequences which we will use for the inferences in the remaining lectures.</p>
</div>
<div class="section" id="inference">
<h3>Inference<a class="headerlink" href="#inference" title="Permalink to this headline">¶</a></h3>
<p>We can understand the inferences that we will try to make from <a class="reference download internal" href="_downloads/Zuckerkandl_and_Pauling-1965.pdf"><tt class="xref download docutils literal"><span class="pre">Zuckerkandl</span> <span class="pre">and</span> <span class="pre">Pauling</span> <span class="pre">(1965)</span></tt></a>:</p>
<p><em>There may thus exist a molecular evolutionary clock.</em></p>
<p>Specifically, we want to estimate how many substitutions occurred (steps in your homework) from the observed sequence divergence.</p>
<p><a class="reference download internal" href="_downloads/Zuckerkandl_and_Pauling-1965.pdf"><tt class="xref download docutils literal"><span class="pre">Zuckerkandl</span> <span class="pre">and</span> <span class="pre">Pauling</span> <span class="pre">(1965)</span></tt></a> present a very crude way to make these inferences. Today we&#8217;ll discuss their approach; next lecture we&#8217;ll start doing better.</p>
</div>
<div class="section" id="id4">
<h3>Inference<a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h3>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/inference.png"><img alt="_images/inference.png" src="_images/inference.png" style="width: 100%;" /></a>
<p class="caption"><a class="reference download internal" href="_downloads/Zuckerkandl_and_Pauling-1965.pdf"><tt class="xref download docutils literal"><span class="pre">Zuckerkandl</span> <span class="pre">and</span> <span class="pre">Pauling</span> <span class="pre">(1965)</span></tt></a></p>
</div>
</div>
<div class="section" id="id5">
<h3>Inference<a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h3>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/inference_plot.png"><img alt="_images/inference_plot.png" src="_images/inference_plot.png" style="width: 80%;" /></a>
<p class="caption"><a class="reference download internal" href="_downloads/Zuckerkandl_and_Pauling-1965.pdf"><tt class="xref download docutils literal"><span class="pre">Zuckerkandl</span> <span class="pre">and</span> <span class="pre">Pauling</span> <span class="pre">(1965)</span></tt></a></p>
</div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Lecture 1</a><ul>
<li><a class="reference internal" href="#lecture-1-the-evolution-of-protein-coding-genes">Lecture 1: the evolution of protein-coding genes</a><ul>
<li><a class="reference internal" href="#genes-evolve-at-the-level-of-dna-or-rna-for-some-viruses">Genes evolve at the level of DNA (or RNA for some viruses)</a></li>
<li><a class="reference internal" href="#but-most-selection-is-on-the-protein">But most selection is on the protein</a></li>
<li><a class="reference internal" href="#properties-of-proteins-are-not-easily-abstracted">Properties of proteins are not easily abstracted</a></li>
<li><a class="reference internal" href="#context-for-reading-zuckerkandl-and-pauling-1965">Context for reading Zuckerkandl and Pauling (1965)</a></li>
<li><a class="reference internal" href="#they-knew-the-structure-of-sperm-whale-myoglobin">They knew the structure of sperm whale myoglobin</a></li>
<li><a class="reference internal" href="#they-knew-that-single-mutations-could-potentially-have-large-biological-consequences">They knew that single mutations could potentially have large biological consequences</a></li>
<li><a class="reference internal" href="#they-knew-the-sequences-of-a-variety-of-homologous-globins">They knew the sequences of a variety of homologous globins</a></li>
<li><a class="reference internal" href="#analysis-of-myoglobin-homologs">Analysis of myoglobin homologs</a></li>
<li><a class="reference internal" href="#myoglobin-homologs">Myoglobin homologs</a></li>
<li><a class="reference internal" href="#site-variability-among-myoglobin-homologs">Site variability among myoglobin homologs</a></li>
<li><a class="reference internal" href="#very-little-sequence-is-strictly-conserved">Very little sequence is strictly conserved</a></li>
<li><a class="reference internal" href="#relatively-few-insertions-and-deletions">Relatively few insertions and deletions</a></li>
<li><a class="reference internal" href="#structure-is-highly-conserved">Structure is highly conserved</a></li>
<li><a class="reference internal" href="#id1">Structure is highly conserved</a></li>
<li><a class="reference internal" href="#different-proteins-evolve-at-different-rates">Different proteins evolve at different rates</a></li>
<li><a class="reference internal" href="#zuckerkandl-and-pauling-propose-a-molecular-clock">Zuckerkandl and Pauling propose a molecular clock</a></li>
<li><a class="reference internal" href="#id2">Zuckerkandl and Pauling propose a molecular clock</a></li>
<li><a class="reference internal" href="#important-implications">Important implications</a></li>
<li><a class="reference internal" href="#quantitative-analyses-of-the-molecular-evolution-of-protein-coding-genes">Quantitative analyses of the molecular evolution of protein-coding genes</a></li>
<li><a class="reference internal" href="#the-evolution-studied-in-molecular-phylogenetics-is-due-to-both-mutation-and-selection">The evolution studied in molecular phylogenetics is due to both mutation and selection</a></li>
<li><a class="reference internal" href="#we-will-ignore-the-intra-population-polymorphism">We will ignore the intra-population polymorphism</a></li>
<li><a class="reference internal" href="#we-assume-that-selection-is-on-the-protein-sequence">We assume that selection is on the protein sequence</a></li>
<li><a class="reference internal" href="#we-assume-that-genes-can-be-aligned-and-ignore-gaps">We assume that genes can be aligned, and ignore gaps</a></li>
<li><a class="reference internal" href="#we-envision-mutation-as-identical-across-sites">We envision mutation as identical across sites</a></li>
<li><a class="reference internal" href="#we-know-that-selection-is-not-identical-across-sites">We know that selection is <strong>not</strong> identical across sites</a></li>
<li><a class="reference internal" href="#id3">We know that selection is <strong>not</strong> identical across sites</a></li>
<li><a class="reference internal" href="#simulating-this-process-your-homework">Simulating this process (your homework)</a></li>
<li><a class="reference internal" href="#inference">Inference</a></li>
<li><a class="reference internal" href="#id4">Inference</a></li>
<li><a class="reference internal" href="#id5">Inference</a></li>
</ul>
</li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="index.html"
                        title="previous chapter">Genome Sciences 541</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/lecture_1.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="index.html" title="Genome Sciences 541"
             >previous</a> |</li>
        <li><a href="index.html">Genome Sciences 541 2014.04.17 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &copy; Copyright 2014, Jesse Bloom.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3a0.
    </div>
  </body>
</html>